commit 5434cc2c4152dddcfb2a6f6cb39b6ff33099a193
Author: Florian Weimer <fweimer@redhat.com>
Date:   Wed Nov 6 10:33:44 2024 +0100

    elf: Do not define consider_profiling, consider_symbind as macros
    
    This avoids surprises when refactoring the code if these identifiers
    are re-used later in the file.
    
    Reviewed-by: DJ Delorie <dj@redhat.com>
    (cherry picked from commit a79642204537dec8a1e1c58d1e0a074b3c624f46)

diff --git a/elf/dl-reloc.c b/elf/dl-reloc.c
index 4bf7aec88b844bc7..b2c1627ceb847486 100644
--- a/elf/dl-reloc.c
+++ b/elf/dl-reloc.c
@@ -220,8 +220,8 @@ _dl_relocate_object (struct link_map *l, struct r_scope_elem *scope[],
   int lazy = reloc_mode & RTLD_LAZY;
   int skip_ifunc = reloc_mode & __RTLD_NOIFUNC;
 
-#ifdef SHARED
   bool consider_symbind = false;
+#ifdef SHARED
   /* If we are auditing, install the same handlers we need for profiling.  */
   if ((reloc_mode & __RTLD_AUDIT) == 0)
     {
@@ -240,9 +240,7 @@ _dl_relocate_object (struct link_map *l, struct r_scope_elem *scope[],
     }
 #elif defined PROF
   /* Never use dynamic linker profiling for gprof profiling code.  */
-# define consider_profiling 0
-#else
-# define consider_symbind 0
+  consider_profiling = 0;
 #endif
 
   /* If DT_BIND_NOW is set relocate all references in this object.  We
@@ -300,7 +298,6 @@ _dl_relocate_object (struct link_map *l, struct r_scope_elem *scope[],
 
     ELF_DYNAMIC_RELOCATE (l, scope, lazy, consider_profiling, skip_ifunc);
 
-#ifndef PROF
     if ((consider_profiling || consider_symbind)
 	&& l->l_info[DT_PLTRELSZ] != NULL)
       {
@@ -321,7 +318,6 @@ _dl_relocate_object (struct link_map *l, struct r_scope_elem *scope[],
 	    _dl_fatal_printf (errstring, RTLD_PROGNAME, l->l_name);
 	  }
       }
-#endif
   }
 
   /* Mark the object so we know this work has been done.  */
